src/lib/viewers/doc/__tests__/DocBaseViewer-test.js

```diff
@@ -1704,38 +1704,57 @@ describe('src/lib/viewers/doc/DocBaseViewer', () => {
-                docBase.loadUIReact();
-
-                expect(docBase.controls).toBeInstanceOf(ControlsRoot);
-                expect(docBase.controls.render).toBeCalledWith(
-                    <DocControls
-                        annotationMode="none"
-                        hasHighlight={false}
-                        hasRegion={false}
-                        maxScale={10}
-                        minScale={0.1}
-                        onAnnotationModeClick={docBase.handleAnnotationControlsClick}
-                        onAnnotationModeEscape={docBase.handleAnnotationControlsEscape}
-                        onFindBarToggle={docBase.toggleFindBar}
-                        onFullscreenToggle={docBase.toggleFullscreen}
-                        onPageChange={docBase.setPage}
-                        onPageSubmit={docBase.handlePageSubmit}
-                        onThumbnailsToggle={docBase.toggleThumbnails}
-                        onZoomIn={docBase.zoomIn}
-                        onZoomOut={docBase.zoomOut}
-                        pageCount={docBase.pdfViewer.pagesCount}
-                        pageNumber={docBase.pdfViewer.currentPageNumber}
-                        scale={1}
-                    />,
-                );
-            });
+            test.each`
+                areNewAnnotationsEnabled | hasAnnotationCreatePermission | hasDrawing | hasRegion | showAnnotationsDrawingCreate
+                ${false}                 | ${false}                      | ${false}   | ${false}  | ${false}
+                ${true}                  | ${true}                       | ${true}    | ${true}   | ${true}
+            `(
+                'should create controls root and render the react controls with hasDrawing set to $hasDrawing and hasRegion set to $hasRegion',
+                ({
+                    areNewAnnotationsEnabled,
+                    hasAnnotationCreatePermission,
+                    hasDrawing,
+                    hasRegion,
+                    showAnnotationsDrawingCreate,
+                }) => {
+                    docBase.options.showAnnotationsDrawingCreate = showAnnotationsDrawingCreate;
+                    docBase.options.useReactControls = true;
+                    docBase.pdfViewer = {
+                        currentPageNumber: 2,
+                        currentScale: 1,
+                        pagesCount: 3,
+                    };
+                    jest.spyOn(docBase, 'areNewAnnotationsEnabled').mockReturnValue(areNewAnnotationsEnabled);
+                    jest.spyOn(docBase, 'hasAnnotationCreatePermission').mockReturnValue(hasAnnotationCreatePermission);
+                    stubs.checkPermission.mockReturnValue(false);
+
+                    docBase.loadUIReact();
+
+                    expect(docBase.controls).toBeInstanceOf(ControlsRoot);
+                    expect(docBase.controls.render).toBeCalledWith(
+                        <DocControls
```

Comment:
Can we keep the original test and add a smaller `test.each` just for these assertions? We shouldn't need to test all of these props so many times.

Reply:
Yeah this test will eventually go away once this feature is released, so it'll be easier to clean up as a separate test